hedera-node/src/main/java/com/hedera/services/state/expiry/PriorityQueueExpiries.java

```diff
@@ -0,0 +1,68 @@
+package com.hedera.services.state.expiry;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import java.util.PriorityQueue;
+
+/**
+ * Priority Queue of expiration events in which events are ordered based on expiry
+ */
+public class PriorityQueueExpiries<K> implements KeyedExpirations<K> {
+	private long now = 0L;
+	private PriorityQueue<ExpiryEvent<K>> allExpiries = new PriorityQueue<>();
+
+	@Override
+	public void reset() {
+		now = 0L;
+		allExpiries.clear();
+	}
+
+	@Override
+	public void track(K id, long expiry) {
+		allExpiries.offer(new ExpiryEvent(id, expiry));
+		now = allExpiries.peek().getExpiry();
+	}
+
+	@Override
+	public boolean hasExpiringAt(long now) {
+		return !allExpiries.isEmpty() && allExpiries.peek().isExpiredAt(now);
```

Comment:
nit: we already have `this.now` right. `peek()` is constant time anyway but since we already keep track of peek's expiry in now we can just compare using that.

Reply:
@anighanta  Here `this.now` is updated when a new entity is inserted into the PriorityQueue. But `hasExpiringAt` is where we give a consensus timestamp as parameter to the method, and see if there any expiration events before that time while purging entities.

Reply:
Ahh, my bad.. I see that we are not updating `now` when removing. Then may be we can remove `now` totally and then just do `allExpiries.isEmpty() ? 0 : allExpiries.peek().getExpiry()` in `getNow()`